### PR TITLE
Song upload

### PR DESCRIPTION
**Pull request by coryj1990**
Had to settle some devise issues, but now looking to see how well a carrierwave uploader will do with our db's.

### CODE REVIEW

**Comment by sumeetjain** [comment]
Please be mindful of unnecessary changes being entered into a PR. This PR has a lot of changes like this:



When you do that, it creates mental noise for whoever is reviewing your code. It also messes with file history data, as it looks superficially like those files had some meaningful changes performed onto them.

**Comment by coryj1990** [comment]
I only did those to relieve some OCD-ish tendencies. I will be sure to remove files like that from the git staging before my next push.

**Comment by sumeetjain** [comment]
Before I look any further at this, please check the 'Files Changes' tab above to see all the new code changes your branch includes. **Anything** that is not necessary for the specific task assigned to you needs to be deleted.

For example, I see controller actions for editing a song. That's not related to this feature at all. Do a _thorough_ audit, and re-submit when the Files Changed tab only contains code I should care about for this feature.

File: app/controllers/songs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  # GET /songs/new
+  def new
+    @song = Song.new
+  end
+
+  # POST /songs
+  # POST /songs.json
+  def create
+    binding.pry
```
Comment (sumeetjain) [inline app/controllers/songs_controller.rb]: Remove all Pry calls before submitting.

File: app/controllers/songs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @song = Song.new(song_params)
+    @song.song_file = params[:file]
```
Comment (sumeetjain) [inline app/controllers/songs_controller.rb]: All of the params for `Song.new` should come from the `song_params` method. You should change your form such that the file is `params[:song][:file]`.

Ask for help from other students (Ask in the Gumiho room). 15 minutes by yourself or 5 minutes of pairing with someone should be enough to solve this issue, **so do not spend 30 minutes trying to get through it on your own if it takes longer than that**.

File: app/controllers/songs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    @song.song_name = params['song']['song_name']
+    @song.album_id = params['song']['album_id']
+    @song.save
+    redirect_to'/songs/new'
```
Comment (sumeetjain) [inline app/controllers/songs_controller.rb]: Should be `redirect_to '/songs/new'`

File: app/controllers/songs_controller.rb (hunk 2 above)
Comment (sumeetjain) [inline app/controllers/songs_controller.rb]: I just noticed that `song_file` is already in song_params. So use that instead of this newly added `file` attribute.

File: spec/controllers/songs_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+      it "re-renders the 'edit' template" do
+        song = Song.create! valid_attributes
+        put :update, params: {id: song.to_param, song: invalid_attributes}, session: valid_session
+        expect(response).to render_template("edit")
+      end
+    end
+  end
+
+  describe "DELETE #destroy" do
```
Comment (sumeetjain) [inline spec/controllers/songs_controller_spec.rb]: There is a lot in this file that is not related to your code. To make things simple: Delete everything from this file that you did not write yourself. If that means deleting the entire file, so be it.

File: app/controllers/songs_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class SongsController < ApplicationController
+
+  # GET /songs/new
+  def new
+    @song = Song.new
+  end
+
+  # POST /songs
+  # POST /songs.json
+  def create
```
Comment (sumeetjain) [inline app/controllers/songs_controller.rb:10]: This entire action should be this:

``` ruby
def create
  @song = Song.new(song_params)
  redirect_to '/songs/new'
end
```

Figure out how the form in your view needs to change to make that possible. Seek help from another student if you want them to help. 15 minutes maximum, or come get help from me.
